### ISSUE TITLE
Investigate user friendly ensembling methods

### ISSUE THREAD

**Issue by RAMitchell** [rapidsai/legate-boost#167]
Design and implement an API where models can be easily combined together. This relies on the additive nature of gradient boosted trees.

This feature would enable more research around ensembling and allow practitioners to easly mix in old models with new instead of retraining from scratch.

For example
```python
# plus operator between models
model_c = model_a + model_b
# scalar multiplication
model_d = model_c * 0.3 + model_a * 0.7
```

We need to be careful here about incompatible model types being added together e.g. a regressor and classifier or a classifier with two different objective types. We can potentially look for different objectives and throw an error.

Other issues include different model parameters e.g. learning_rate. Do we just take the parameter of the first model? Do we throw a warning if they are different?
